mfai/pytorch/losses/gan_dgmr.py

```diff
@@ -0,0 +1,74 @@
+"""Module for various loss functions used with DGMR GAN."""
+
+import torch
+from torch import Tensor, nn
+from torch.nn import functional as F
+
+
+class GridCellLoss(nn.Module):
+    """Grid Cell Regularizer loss from Skillful Nowcasting, see https://arxiv.org/pdf/2104.00954.pdf."""
+
+    def __init__(
+        self,
+        precip_weight_cap: float = 24.0,
+    ) -> None:
+        """
+        Initialize GridCellLoss.
+
+        Args:
+            weight_fn: A function to compute weights for the loss.
+            precip_weight_cap: Custom ceiling value for the weight function.
+        """
+        super().__init__()
+        self.precip_weight_cap = precip_weight_cap
+
+    def forward(self, generated_images: Tensor, targets: Tensor) -> Tensor:
+        """
+        Forward function.
+
+        Calculates the grid cell regularizer value, assumes generated images are the mean
+        predictions from 6 calls to the generator (Monte Carlo estimation of the
+        expectations for the latent variable)
+
+        Args:
+            generated_images: Mean generated images from the generator
+            targets: Ground truth future frames
+
+        Returns:
+            Grid Cell Regularizer term
+        """
+        difference = generated_images - targets
+        if self.weight_fn is not None:
+            weights = self.weight_fn(targets, self.precip_weight_cap)
+            difference = difference * weights
```

Comment:
May be use in-place operation ?
```suggestion
            difference *= weights
```